### PR TITLE
Fix for  Storage cluster creation failed with 'Cannot read property "protocol" of null ' on adding& then cancelling Encryption 

### PR DESCRIPTION
**Pull request by GowthamShanmugam**
Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1910483

Signed-off-by: Gowtham Shanmugasundaram <gshanmug@redhat.com>

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx [hunk 1]
```diff
@@ -76,10 +76,14 @@ export const EncryptionFormGroup: React.FC<EncryptionFormGroupProps> = ({
   }, [encryption.clusterWide, encryption.storageClass, encryptionChecked]);
 
   const toggleEncryption = (checked: boolean) => {
-    setEncryptionDispatch(ActionType.SET_ENCRYPTION, mode, dispatch, {
+    const encryptOj = {
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx:79]:
```suggestion
    const payload = {
```

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx [hunk 2]
```diff
@@ -76,10 +76,14 @@ export const EncryptionFormGroup: React.FC<EncryptionFormGroupProps> = ({
...
       ...encryption,
       clusterWide: checked,
-    });
+    }
+    if(!checked){
+      encryptOj.advanced = false;
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx:84]:
```suggestion
      payload.advanced = false;
```

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx [hunk 3]
```diff
@@ -76,10 +76,14 @@ export const EncryptionFormGroup: React.FC<EncryptionFormGroupProps> = ({
...
+    }
+    setEncryptionDispatch(ActionType.SET_ENCRYPTION, mode, dispatch, encryptOj);
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx:86]:
```suggestion
    setEncryptionDispatch(ActionType.SET_ENCRYPTION, mode, dispatch, payload);
```

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx (hunk 2 above)
Comment (cloudbehl) [inline frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx:84]: why you want to set `advance encryption` to `true` when checked is set to `false`?

**Review by afreen23 — CHANGES_REQUESTED** [review]
/ok-to-test
Nit: Some naming suggestions.

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx (hunk 1 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx:79]: ack

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx (hunk 2 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx:84]: ack

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx (hunk 3 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx:86]: ack

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx (hunk 2 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx:84]: I am setting advance encryption false only when Encryption is unchecked or set to false

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx [hunk 4]
```diff
@@ -76,10 +76,14 @@ export const EncryptionFormGroup: React.FC<EncryptionFormGroupProps> = ({
...
 
   const toggleEncryption = (checked: boolean) => {
-    setEncryptionDispatch(ActionType.SET_ENCRYPTION, mode, dispatch, {
+    const payload = {
       ...encryption,
       clusterWide: checked,
-    });
+    }
+    if(!checked){
+      payload.advanced = false;
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx:84]: We should also set all encryption level and both encryption and kms states to initial value when the high level encryption box is unchecked.
This is to ensure that stale states dont cause errors,
```
      payload.advanced = false;
      payload.storageClass = false;
      payload.hasHandled = false;
```
There is already one check for clearing kms state on L88, please merge the above with this one.
https://github.com/openshift/console/blob/c29d7499e8e44868bfa33d4621fa7875cc696980/frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx#L88-L91

**Review by afreen23 — CHANGES_REQUESTED** [review]

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx [hunk 5]
```diff
@@ -76,10 +76,14 @@ export const EncryptionFormGroup: React.FC<EncryptionFormGroupProps> = ({
   }, [encryption.clusterWide, encryption.storageClass, encryptionChecked]);
 
   const toggleEncryption = (checked: boolean) => {
-    setEncryptionDispatch(ActionType.SET_ENCRYPTION, mode, dispatch, {
+    const payload = {
       ...encryption,
       clusterWide: checked,
-    });
+    }
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx:82]:
```suggestion
    };
```

Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx:82]: You are getting formatting errors hence tests are failing:
```
 /go/src/github.com/openshift/console/frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx
  82:6  error  Insert `;`                                prettier/prettier
  83:7  error  Replace `(!checked)` with `·(!checked)·`  prettier/prettier 
  ```

**Review by afreen23 — CHANGES_REQUESTED** [review]

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx (hunk 4 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx:84]: ack

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx (hunk 5 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx:82]: Let me check this

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx [hunk 6]
```diff
@@ -76,14 +76,18 @@ export const EncryptionFormGroup: React.FC<EncryptionFormGroupProps> = ({
...
-    setEncryptionDispatch(ActionType.SET_ENCRYPTION, mode, dispatch, {
+    const payload = {
       ...encryption,
       clusterWide: checked,
-    });
-    setEncryptionChecked(checked);
+    };
     if (!checked) {
+      payload.advanced = false;
+      payload.hasHandled = false;
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx:85]: That seems weird but it is used for validation and need to be set to `true`
```suggestion
      payload.hasHandled = true;
```

**Review by afreen23 — CHANGES_REQUESTED** [review]

**Review by afreen23 — COMMENTED** [review]
Squash the commits as well.

**Comment by GowthamShanmugam** [comment]
@afreen23 test case passed, Please review

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx (hunk 6 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx:85]: let me check this

Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx:85]: The initial state sets "hasHandled" to true, lets follow the same defaults.

**Comment by cloudbehl** [comment]
@GowthamShanmugam can you squash the commits.

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx [hunk 7]
```diff
@@ -76,14 +76,17 @@ export const EncryptionFormGroup: React.FC<EncryptionFormGroupProps> = ({
   }, [encryption.clusterWide, encryption.storageClass, encryptionChecked]);
```
Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/src/components/ocs-install/install-wizard/configure.tsx:76]: I can see a separate useEffect is to set hasHandled for all cases, Setting a flag is kind of redundant and leads to unpredicted behavior, so I removed it from my latest change

**Comment by GowthamShanmugam** [comment]
@cloudbehl @afreen23 all changes are done, please review

**Pull request merged**